src/halo-hooks/amm/useSwapToken.ts

```diff
@@ -0,0 +1,221 @@
+import { useCallback, useEffect, useState } from 'react'
+import ROUTER_ABI from 'constants/haloAbis/Router.json'
+import ASSIMILATOR_ABI from 'constants/haloAbis/Assimilator.json'
+import { formatUnits, parseUnits } from 'ethers/lib/utils'
+import { ethers } from 'ethers'
+import { getContract } from 'utils'
+import { ChainId, CurrencyAmount, Token } from '@sushiswap/sdk'
+import { AssimilatorAddressMap, haloAssimilators, haloUSDC, TokenSymbol } from 'constants/tokenLists/halo-tokenlist'
+import { useTransactionAdder } from 'state/transactions/hooks'
+import { useActiveWeb3React } from '../../hooks'
+import { ERC20_ABI } from 'constants/abis/erc20'
+import useCurrentBlockTimestamp from 'hooks/useCurrentBlockTimestamp'
+import { ChainAddressMap } from '../../constants'
+import { toFixed } from 'utils/formatNumber'
+
+const routerAddress: ChainAddressMap = {
+  [ChainId.MAINNET]: '',
+  [ChainId.KOVAN]: '0xa02dCeB15cc32249beC33C2808b4799a44F8B0D5',
+  [ChainId.MATIC]: '0x26f2860cdeB7cC785eE5d59a5Efb2D0D3842C39D'
+}
+
+export enum CurrencySide {
+  TO_CURRENCY = 'toCurrency',
+  FROM_CURRENCY = 'fromCurrency'
+}
+
+export const useSwapToken = (toCurrency: Token, fromCurrency: Token) => {
+  const { account, chainId, library } = useActiveWeb3React()
+  const [price, setPrice] = useState<number>()
+  const currentBlockTime = useCurrentBlockTimestamp()
+  const [minimumAmount, setMinimumAmount] = useState<string | undefined>()
+  const [allowance, setAllowance] = useState('0')
+  const addTransaction = useTransactionAdder()
+
+  const toSafeValue = useCallback(
+    (value: string) => {
+      if (value.charAt(value.length - 1) === '.') {
+        return value.substring(0, value.length - 1)
+      } else {
+        return toFixed(Number(value), fromCurrency.decimals)
+      }
+    },
+    [fromCurrency.decimals]
+  )
+
+  const toSafeCurrencyValue = (value: CurrencyAmount, currencySide: CurrencySide) => {
+    return value.toFixed(currencySide === CurrencySide.FROM_CURRENCY ? fromCurrency.decimals : toCurrency.decimals)
+  }
+
+  const getFutureTime = useCallback(
+    (addMinutes: number) => {
+      if (currentBlockTime) {
+        return currentBlockTime.add(addMinutes).toNumber()
+      } else {
+        return new Date().getTime() + addMinutes
+      }
```

Comment:
We could remove the else :)

```suggestion
      if (currentBlockTime) {
        return currentBlockTime.add(addMinutes).toNumber()
      }
      return new Date().getTime() + addMinutes
```

Reply:
applying this change on your hook @schystz